.github/workflows/build-test-release.yml

```diff
@@ -0,0 +1,27 @@
+name: Build, Test and Release
+on: [push]
+jobs:
+  build-test:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+          cache: 'yarn'
+      - run: yarn install
+      - run: yarn build
+      - run: yarn test
+      - id: publish
+        name: Release to NPM
+        if: github.ref == 'refs/heads/master'
+        uses: JS-DevTools/npm-publish@v1
+        with:
+          token: ${{ secrets.NPM_TOKEN }}
```

Comment:
A valid NPM_TOKEN secret is required to be setup in this repository for this to work.

Reply:
Would this publish on npm using beta version too?

Reply:
It should use the version in the package.json file as the `latest` release version when publishing. Or do you mean something else?

Reply:
I've included the beta tag now.